hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,285 @@ title: "OTP Controller Technical Specification"
+
+This verification ensures the value stored in the buffer registers remain consistent with those in the OTP.
+This process re-reads the OTP at semi-random intervals and confirms the value read is the same as the value stored.
+Note, given there are integrity checks in parallel, it is not necessary for some partitions to check ALL read contents for consistency.
+If there is an integrity digest, only the digest needs to be read; otherwise, all values must be read.
+
+
+This check applies to LIFE_CYCLE, HW_CFG and SECRET partitions.
+If a failure is encountered, the OTP controller will send out an `otp_consistency_mismatch` alert and reset all of its hardware outputs to their defaults (**TBD do we need an LFSR clearing mechanism here for the secret partition?**)
+
+Note that checks applied to life cycle could cause a failure if life cycle is updated, because life cycle is the only partition that may contain live updates.
+The controller hence detects this condition and makes sure that the buffer registers are kept up to date in order to prevent false positives.
+
+### Secret Partition Integrity Checks
+
+Since the secret partition is stored encrypted, this also implies the integrity digest is calculated over the encrypted form.
+In order to balance the amount of buffer registers needed, only the decrypted form of the secret partition is held in buffer registers.
+Hardware calculates the digest by re-encrypting the data before passing it through the digest.
+
+
+## Power-up and Sense
+
+The OTP controller partition storage must output a specified default (it is not always 0 like a blank OTP) upon reset release.
+This default output must remain until the OTP controller completes all checks.
+
+The OTP controller reads from the OTP IP.
+If the reads pass OTP IP internal checks (for example ECC or redundancy), the partition storage is updated; however the output is still held at the default state via an output mux.
+After all read is complete, the OTP controller performs integrity checks on the HW_CFG and SECRET partitions.
+If a partition fails the integrity checks at this point it would signal an initialization error in the status CSR and abort further initialization.
+
+After all integrity checks are complete, the OTP controller releases the output gating and marks outputs as valid.
+However, any partition marked with "error" continues to hold its output in the default state.
+
+Once the above steps are complete, the partition storage in buffered registers is not updated again (except for updates to the life cycle partition through the life cycle interface).
+I.e., values programmed to OTP via the programming interface will not be visible in buffered registers until after the next power cycle.
+
+At this point, outputs of the partition storage are NOT expected to change unless a periodic check suddenly fails.
+When this failure occurs, all outputs are reverted to its default state, and an alert is immediately triggered to the alert manager.
```

Comment:
shall we call it alert handler instead?

Reply:
yes that would be more consistent. I'll change that.